use_case_test.go

```diff
@@ -0,0 +1,78 @@
+package es_test
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/indebted-modules/es"
+	"github.com/stretchr/testify/suite"
+)
+
+type UseCaseSuite struct {
+	suite.Suite
+}
+
+func TestUseCaseSuite(t *testing.T) {
+	suite.Run(t, new(UseCaseSuite))
+}
+
+func (s *UseCaseSuite) TestAggregateLifecycle() {
+	store := es.NewStore(es.NewInMemoryDriver())
+
+	sampleAggregate := &SampleAggregate{}
+	events := sampleAggregate.DoSomething("1", []string{"event-1", "event-2"})
+	err := store.Save(events)
+	s.NoError(err)
+	s.Equal(
+		[]string{"event-1", "event-2"},
+		sampleAggregate.ReducedData,
+		"Apply events during memory lifecycle",
+	)
+
+	newInMemoryCycle := &SampleAggregate{}
+	err = store.Load("1", newInMemoryCycle)
+	s.NoError(err)
+	s.Equal(
+		[]string{"event-1", "event-2"},
+		newInMemoryCycle.ReducedData,
+		"Re-apply events when loading aggregate to memory again",
+	)
+}
+
+func (s *UseCaseSuite) TestConcurrentWriting() {
```

Comment:
This is the only test that test we're setting the aggregate version in the store

Reply:
https://github.com/indebted-modules/es/blob/master/store.go#L27